src/linux/DisplayInfoLinux.cpp

```diff
@@ -10,16 +14,21 @@ using namespace VSTGUI;
     
 float getDisplayBackingScaleFactor(CFrame *)
 {
-    Surge::UserInteractions::promptError("getDisplayBackingScaleFactor not implemented yet on linux.",
-                                         "Software Error");
+    std::cerr << "BackingDisplayFactor is not implemented on linux. Returning 1.0" << std::endl;
     return 1.0;
 }
     
 CRect getScreenDimensions(CFrame *)
 {
-    Surge::UserInteractions::promptError("getScreenDimensions not implemented yet on linux. Returning 1400x1050",
-                                         "Software Error");
-    return CRect(CPoint(0,0), CPoint(1400,1050)); // a typical mid-range 15" laptop
+    /*
+    ** For now make the assumption of single screen to at least avoid
+    ** clip and bound errors like shown in #594
+    **
+    ** FIXME: Deal with multiple screens using the appropriate API
+    */
+    Display* d = XOpenDisplay(NULL);
```

Comment:
This is leaking memory/resources everytime it is called, it needs the cleanup methods

Reply:
Thanks. That’s just xclosedisplay? Or do I need to dispose of the screen also

Reply:
Googling shows me xclosedisplay is the call I need. I’ll do that. But also I will cache this value since we don’t support multi screen anyway in this revision and will update the fixme